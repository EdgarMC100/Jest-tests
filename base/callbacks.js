export function callbackHell(callback){
    callback('Hi')
}