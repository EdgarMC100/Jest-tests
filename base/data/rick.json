{
    "id": 1,
    "name": "Rick Sanchez",
    "status": "Dead",
    "species": "Human",
    "gender": "Male"
}