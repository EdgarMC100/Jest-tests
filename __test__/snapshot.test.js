import { getCharacter } from "../base/snapshot";
import rick from '../base/data/rick.json'

describe('Es hora de las instantaneas', () => {
    it('Should be a Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    })
    it('Deberia fallar la instantanea', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20)
        }
        expect(user).toMatchSnapshot();
    })
    
    it('Should be have an exception inside the code', () => {
        const user = {
            createdAt: new Date(),
            id: Math.floor(Math.random() * 20),
            name: 'LeBron James',
          };
        
          expect(user).toMatchSnapshot({
            createdAt: expect.any(Date),
            id: expect.any(Number),
          });
    })
});