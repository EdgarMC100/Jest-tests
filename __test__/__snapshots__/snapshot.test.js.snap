// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Deberia fallar la instantanea 1`] = `
Object {
  "createAt": 2021-07-22T15:07:01.508Z,
  "id": 1,
}
`;

exports[`Es hora de las instantaneas Should be a Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Dead",
}
`;

exports[`Es hora de las instantaneas Should be have an exception inside the code 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;
